projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -1003,9 +1003,9 @@ export class IgxColumnComponent implements AfterContentInit {
     }
 
     get rightPinnedOffset(): string {
-        return this.pinned ?
+        return (this.pinned && !this.grid.isPinningToStart) ?
             - this.grid.pinnedWidth + 'px' :
-            null;
+            this.grid.isPinningToStart ? 0 + 'px' : null;
```

Comment:
@3phase I think if pinning is to start, it would be better to leave it null, it is not necessary to apply the style left 0px for the cells that don't need it